src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.DefaultProperties.targets

```diff
@@ -24,10 +24,13 @@
       See: https://github.com/dotnet/sdk/blob/955c0fc7b06e2fa34bacd076ed39f61e4fb61716/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L16
     -->
     <_GetChildProjectCopyToPublishDirectoryItems>false</_GetChildProjectCopyToPublishDirectoryItems>
-    <UseMonoRuntime Condition=" '$(PublishAot)' == 'true' and '$(UseMonoRuntime)' == '' ">false</UseMonoRuntime>
+    <!-- Define a $(_AndroidNativeAot) property, as the logic detecting NativeAOT may change in the future -->
+    <_AndroidNativeAot Condition=" '$(PublishAot)' == 'true' ">true</_AndroidNativeAot>
+    <_AndroidNativeAot Condition=" '$(_AndroidNativeAot)' == '' ">false</_AndroidNativeAot>
+    <UseMonoRuntime Condition=" '$(_AndroidNativeAot)' == 'true' and '$(UseMonoRuntime)' == '' ">false</UseMonoRuntime>
```

Comment:
What happens/what *should* happen when `$(PublishAot)`=true and `$(UseMonoRuntime)`=true?  As `$(UseMonoRuntime)` is a public property, there is potential expectation that users could do this, so what should happen?

Perhaps we should warn?

Reply:
Customers usually would not set `$(UseMonoRuntime)`, but they would set `$(PublishAot)` to opt into NativeAOT.

iOS has the following logic in .NET 9, that is up for discussion:

* `dotnet build` uses Mono, unless you specify `-p:_IsPublishing=true`
* `dotnet publish` uses NativeAOT, both Debug & Release

I don't think we should change anything, until we know what it *should do* and align with iOS.

I kind of feel like `Debug` mode should use Mono and `Release` is NativeAOT. No need to use a private `$(_IsPublishing)` property or `dotnet publish`.